IBAnimatable/BackgroundImageDesignable.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Created by phimage on 25/03/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import Foundation
+
+import UIKit
+
+/// Protocol for designing background image
+public protocol BackgroundImageDesignable {
+
+  /**
+   * The background image
+   */
+  var backgroundImage: UIImage? { get set }
+
+}
+
+/// Protocol for designing background
+public protocol BackgroundDesignable: class {
+
+  /**
+   * The background view
+   */
+  var backgroundView: UIView? { get set }
+
+}
+
+extension UITableViewCell: BackgroundDesignable {}
+extension UITableViewHeaderFooterView: BackgroundDesignable {}
+extension UITableView: BackgroundDesignable {}
+extension UICollectionViewCell: BackgroundDesignable {}
+extension UICollectionView: BackgroundDesignable {}
+
+fileprivate typealias BackgroundImageView = AnimatableImageView
```

Comment:
agree with @tbaranes for using `AnimatableImageView` instead of `typealias BackgroundImageView`. 

On the other hand for the private subclass. I think creating a private subclass is good for some case I can think of. For example, we can check like `backgroundView as? PrivateImageView` then we know `backgroundView` must be set within `extension BackgroundImageDesignable`. Another someone may use `AnimatableImageView` for `backgroundView` outside of `IBAnimatable` framework. And we should not  reset `backgroundView = nil` if `backgroundView` is not original set by `extension BackgroundImageDesignable`.

Is it the case you are thinking of, @phimage ? I leave it to you to make a decision to use `private class PrivateAnimatableImageView: AnimatableImageView` or not.